lib/bin/supllib/include/utils.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#ifndef SUPL_UTILS_H_
+#define SUPL_UTILS_H_
+
+/**
+ * @brief Convert hex string into hex
+ *        Example: in = {'D', 'E', 'A', 'D'} -> out = {0xDE, 0xAD}
+ *
+ * @param[in]   in        Hex string buffer to convert
+ * @param[in]   in_len    Input buffer length
+ * @param[out]  out       Hex buffer for the output
+ * @param[in]   out_len   Hex buffer length, shall be at least (in_len / 2)
+ *
+ * @return  hex output length on success
+ *         -1 on error
+ */
+int hexstr2hex(const char *restrict in,
```

Comment:
This code I copied from the orginal code
https://www.geeksforgeeks.org/restrict-keyword-c/

Seems to be an optimization hint to the compiler, I can remove it if you want.

Reply:
Could this cause issues for apps that use C++?

Reply:
To be honest, I have learned about this keyword just today so I can't tell from my own experience, but indeed, this bullet point is daunting:
```
* restrict is not supported by C++. It is a C only keyword.
```
Perhaps it would be safer to give up on it if it's really just a copy-paste from some other code.

Reply:
This code was written by the thirdpary consultant company that have done GPS work for us. So by copying this code I mean copying form the internal codebase that they have worked on.

Reply:
Understood, but assuming that app C++ sources could directly or indirectly include utils.h, then this could cause issues.

Reply:
Perhaps you could check that we don't accidentally leak utils.h to the app?

Reply:
We do not need this keyword, it is strictly an optimization hint to the compiler. This seems like a simpler solution to avoid possible issues.

Reply:
Great, let's remove it then.

Reply:
gone